celery_utils/logged_task.py

```diff
@@ -0,0 +1,34 @@
+"""
+Improved logging for celery tasks.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from celery import Task
+
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=abstract-method
+class LoggedTask(Task):
+    """
+    Task base class that emits a log statement when it gets submitted.
+    """
+
+    abstract = True
+
+    def apply_async(self, args, kwargs, **options):
+        """
+        Emit a log statement when the task is submitted.
+        """
+        result = super(LoggedTask, self).apply_async(args, kwargs, **options)
+        log.info('Task {}[{}] submitted with arguments {}, {}'.format(
+            self.name,
+            result.id,
+            args,
+            kwargs
+        ))
```

Comment:
Reviewers, note this has to get run after super().apply_async() gets called, so that the task_id is available.  This does what we want in an async context (logging when the task is sent plus several milliseconds), but if is_eager == True, it doesn't emit this log until the event returns.  I don't think that's a use case we care about enough to try to fix, though.